prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -141,4 +147,39 @@ class FHIRTranslator(
 
         return hl7Message
     }
+
+    /**
+     * Removes observations from a [bundle] that are not referenced in [receiverEndpoint]
+     *
+     * @return [Bundle] with the unwanted observations removed
+     */
+    internal fun removeUnwantedConditions(bundle: Bundle, receiverEndpoint: Endpoint): Bundle {
```

Comment:
Have we discussed how we should organize this code? For example, should each Resource? have it's own FHIRTranslator file? I can see this file potentially getting pretty big if we just keep adding non-generic stuff here.